### PR TITLE
Arbitrary DER ser- and deserialization from a TaggedDerValue type

### PR DESCRIPTION
**Pull request by jethrogb**
This partially reverts #9 and replaces it with new functionality. `write_tagged_der` can now set the appropriate PC bit value without having to poke through the abstraction.

cc @aandyl

### CODE REVIEW

File: src/writer/mod.rs [hunk 1]
```diff
@@ -958,48 +954,32 @@ impl<'a> DERWriter<'a> {
...
-            implicit_tag: Some(tag),
-            override_pc: self.override_pc,
-        };
+        let mut writer = DERWriter::from_buf(self.buf);
+        writer.implicit_tag = Some(tag);
         return callback(writer);
     }
 
-    /// Overrides the primitive/constructed bit on the next value
+    /// Writes the arbitrary tagged DER value in `der`.
```
Comment (est31) [inline src/writer/mod.rs:962]: I think the word here needs to be `arbitrarily` but I'm no native speaker.

File: src/models/der.rs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        }
+    }
+
+    pub fn as_str(&self) -> Option<&str> {
+        use std::str::from_utf8;
+        // RFC 5280 §4.1.2.6.  Subject
+        //
+        // TeletexString, BMPString, and UniversalString are included
+        // for backward compatibility, and SHOULD NOT be used for
+        // certificates for new subjects.
```
Comment (est31) [inline src/models/der.rs]: This comment should be removed as yasna.rs is general-purpose. The code itself is fine though.

**Review by est31 — CHANGES_REQUESTED** [review]
In general the PR is great, but had two remarks.

File: src/writer/mod.rs (hunk 1 above)
Comment (jethrogb) [reply src/writer/mod.rs:962]: Both are correct but have different parse trees.
* (arbitrary (tagged (DER value)))
* ((arbitrarily tagged) (DER value))

Comment (est31) [reply src/writer/mod.rs:962]: Thanks, it's fine then :)

File: src/models/der.rs (hunk 2 above)
Comment (jethrogb) [reply src/models/der.rs]: Done

**Review by est31 — APPROVED** [review]

**Pull request merged**